submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,156 @@
+        [array[i], array[j]] = [array[j], array[i]];
+      };
+    };
+    
+    shuffle(numbersArray);
+    
+    if (document.documentElement.clientHeight > document.documentElement.clientWidth) {
+      if (sizeSelector.value == 12) {
+        gameArea.classList.add("game_area-3columns");
+      } else {
+        gameArea.classList.add("game_area-4columns");
+      };
+    } else {
+      if (sizeSelector.value <= 16) {
+        gameArea.classList.add("game_area-4columns");
+      } else if (sizeSelector.value == 20){
+        gameArea.classList.add("game_area-5columns");
+      } else {
+        gameArea.classList.add("game_area-6columns");
+      };
+    };
+    let fragment = document.createDocumentFragment();
+    for (let i=0; i < sizeSelector.value; i++) {
+      let card = document.createElement("button");
+      card.classList.add("card");
+      card.innerHTML = `<div class="card__front" style="background-image: url(img/${cardThemeSelected.value}_bg.jpg);"></div><div class="card__back" style="background-image: url(img/${cardThemeSelected.value}_${numbersArray[i] + 1}.jpg);"></div>`;
+      fragment.append(card);
+      card.addEventListener("click", checkCard);
+    };
+    gameArea.append(fragment);
+    cardsArray = Array.from(document.querySelectorAll(".card"));
+  };
+  
+  function checkCard({target}) {
+    flipCard.call(target);
+    setTimeout(function() {
+      if (!cardSelected) {
+        cardSelected = cardsArray.indexOf(target) + 1;
+        lockCard.call(target);
+      } else if ((numbersArray[cardSelected - 1] == numbersArray[cardsArray.indexOf(target)])&&(cardSelected - 1 != cardsArray.indexOf(target))) {
+        vanishCard.call(cardsArray[cardSelected - 1]);
+        vanishCard.call(target);
+        cardSelected = "";
+        matchedCounter++;
+        if (matchedCounter == sizeSelector.value/2) {
+          hailWinner();
+        };
+      } else {
+        flipCard.call(cardsArray[cardSelected - 1]);
+        unlockCard.call(cardsArray[cardSelected - 1])
+        cardSelected = "";      
+        flipCard.call(target);
+      };
+    }, 400);
+    
+    function vanishCard() {
+      this.classList.add("card-vanish");
+      this.setAttribute("tabindex", "-1");
+    };
+    
+    function flipCard() {
+      this.firstElementChild.classList.toggle("card__front-flip");
+      this.lastElementChild.classList.toggle("card__back-flip");
+    };
+    
+    function lockCard() {
+      this.style.pointerEvents = "none";
+      this.setAttribute("disabled", "true");
+    };
+    
+    function unlockCard() {
+      this.style.pointerEvents = "auto";
+      this.removeAttribute("disabled");
+    };
+  };
+
+  function hailWinner() {
+    const hailWindow = document.createElement("div");
+    hailWindow.classList.add("popup_hail__background");
+    hailWindow.innerHTML = `<div class="popup_hail__window">Вы выиграли!<button class="popup_hail__button">Ok</button></div>`;
+    doc.append(hailWindow);
```

Comment:
This wrap we can put in tag `main` and does not redrawing.

Reply:
Yes, we can.
But: if user never wins the game - he (she) never needs to see this window, so we don't need to make HTML larger and make the page longer to load :)
At the final stage of working (when I write JS) I always intent to do everything with JS :)
Whell, I can remake it.

Reply:
Sooner or later the player will still win))